src/ILLink.RoslynAnalyzer/RequiresAnalyzerBase.cs

```diff
@@ -187,6 +239,31 @@ private void ReportRequiresDiagnostic (OperationAnalysisContext operationContext
 				url));
 		}
 
+		private void ReportMismatchInAttributesDiagnostic (SymbolAnalysisContext symbolAnalysisContext, ISymbol member, ISymbol baseMember)
+		{
+			string message = string.Empty;
+			string var0 = RequiresAttributeName;
+			string var1 = member.ToDisplayString ();
+			string var2 = baseMember.ToDisplayString ();
+			if (!member.HasAttribute (RequiresAttributeName) && baseMember.IsVirtual)
+				message = string.Format (SharedStrings.BaseRequiresMismatchMessage, var0, var1, var2);
+			else if (member.HasAttribute (RequiresAttributeName) && baseMember.IsVirtual)
+				message = string.Format (SharedStrings.DerivedRequiresMismatchMessage, var0, var1, var2);
+			else if (!member.HasAttribute (RequiresAttributeName) && !baseMember.IsVirtual)
+				message = string.Format (SharedStrings.InterfaceRequiresMismatchMessage, var0, var1, var2);
+			else if (member.HasAttribute (RequiresAttributeName) && !baseMember.IsVirtual)
+				message = string.Format (SharedStrings.ImplementationRequiresMismatchMessage, var0, var1, var2);
+			if (string.IsNullOrEmpty (message)) {
```

Comment:
Instead of checking the value, just use `else`, and don't assign `message` at the beginning of the method, so definite assignment will catch if we forget to set it.